### ISSUE TITLE
Add support for status of jobs through Treeherder instead of BuildAPI

### ISSUE THREAD

**Issue by armenzg** [armenzg/mozilla_ci_tools#156]
mozci [1] currently uses buildapi [2] to determine information about jobs that are displayed on treeherder [3].
However, not every job that is displayed on treeherder was a job that was run through buildapi/buildbot (Releng's CI).
These days we're having jobs being submitted to treeherder from other systems and we will eventually need to feed from the treeherder API [4].
The other advantage of feeding from treeherder is that (I think) we will remove the need for credentials to simply do "read" requests like job status. This would mean that people without credentials could run mozci without needing credentials; unless they actually need to trigger a job.
In other words, people could use mozci for testing purposes.
Another good thing of treeherder is that it can be easily be setup locally (unlike buildapi).
I don't know if their APIs are all implemented or if they will change but it's better to have the support.

[1] https://github.com/armenzg/mozilla_ci_tools
[2] https://secure.pub.build.mozilla.org/buildapi/self-serve
[3] http://treeherder.mozilla.org
[4] http://treeherder-dev.allizom.org/docs

**Comment by armenzg** [comment]
We can close this issue against https://github.com/armenzg/mozilla_ci_tools/issues/178
